aws/resource_aws_vpc_peering_connection_accepter_test.go

```diff
@@ -9,48 +11,176 @@ import (
+					resource.TestCheckResourceAttr(resourceNameAccepter, "peer_region", region),
+					resource.TestCheckResourceAttr(resourceNameAccepter, "accept_status", "active"),
 				),
 			},
 		},
 	})
 }
 
-func TestAccAWSVPCPeeringConnectionAccepter_differentRegion(t *testing.T) {
-	var connection ec2.VpcPeeringConnection
+func TestAccAWSVPCPeeringConnectionAccepter_differentRegionSameAccount(t *testing.T) {
+	var connectionMain, connectionPeer ec2.VpcPeeringConnection
+	var providers []*schema.Provider
+	resourceNameMainVpc := "aws_vpc.main"                              // Requester
+	resourceNamePeerVpc := "aws_vpc.peer"                              // Accepter
+	resourceNameConnection := "aws_vpc_peering_connection.main"        // Requester
+	resourceNameAccepter := "aws_vpc_peering_connection_accepter.peer" // Accepter
+	mainRegion := "us-west-2"                                          // Requester
+	peerRegion := "us-east-1"                                          // Accepter
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccMultipleRegionsPreCheck(t)
+		},
+		ProviderFactories: testAccProviderFactories(&providers),
+		CheckDestroy:      testAccAwsVPCPeeringConnectionAccepterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsVPCPeeringConnectionAccepterConfigDifferentRegionSameAccount(mainRegion, peerRegion),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSVpcPeeringConnectionExistsWithProvider(resourceNameConnection, &connectionMain, testAccAwsRegionProviderFunc(mainRegion, &providers)),
+					testAccCheckAWSVpcPeeringConnectionExistsWithProvider(resourceNameAccepter, &connectionPeer, testAccAwsRegionProviderFunc(peerRegion, &providers)),
+					resource.TestCheckResourceAttrPair(resourceNameConnection, "vpc_id", resourceNameMainVpc, "id"),
+					resource.TestCheckResourceAttrPair(resourceNameConnection, "peer_vpc_id", resourceNamePeerVpc, "id"),
+					resource.TestCheckResourceAttrPair(resourceNameConnection, "peer_owner_id", resourceNamePeerVpc, "owner_id"),
+					resource.TestCheckResourceAttr(resourceNameConnection, "peer_region", peerRegion),
+					// ** TODO See TestAccAWSVPCPeeringConnectionAccepter_sameRegion()
+					// resource.TestCheckResourceAttrPair(resourceNameAccepter, "vpc_id", resourceNamePeerVpc, "id"),
+					// resource.TestCheckResourceAttrPair(resourceNameAccepter, "peer_vpc_id", resourceNameMainVpc, "id"),
+					resource.TestCheckResourceAttrPair(resourceNameAccepter, "peer_owner_id", resourceNameMainVpc, "owner_id"),
+					resource.TestCheckResourceAttr(resourceNameAccepter, "peer_region", peerRegion),
+					resource.TestCheckResourceAttr(resourceNameAccepter, "accept_status", "active"),
+				),
+			},
+		},
+	})
+}
 
+func TestAccAWSVPCPeeringConnectionAccepter_sameRegionDifferentAccount(t *testing.T) {
+	key := "VPC_PEER_AWS_ACCESS_KEY_ID"
```

Comment:
You're definitely on the right track here, but I would like to call out that for cross-account testing we are trying to standardize on a setup like the one that can be found in https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_guardduty_invite_accepter_test.go -- the gist being:

* Test `PreCheck` includes `testAccAlternateAccountPreCheck(t)` to ensure a standardized `AWS_ALTERNATE_ACCESS_KEY_ID` + `AWS_ALTERNATE_SECRET_ACCESS_KEY` or `AWS_ALTERNATE_PROFILE` set of information is provided for cross-account testing credentials
* `ProviderFactories` is used instead of `Providers` like it is here 👍 
* Test configurations pull in a standardized `testAccAlternateAccountProviderConfig()` and use the `aws.alternate` provider alias for the necessary cross-account resources. The resource that is being tested should not use the provider alias to simplify the testing setup.
* Any import testing steps include the `Config` to match the previous step

That being said, these particular acceptance tests are also dealing with same account different region and different account different region testing. The framework mentioned above needs some extensions to support standardized region selection as well, the preference being an optional `AWS_ALTERNATE_REGION` environment variable in conjunction with two new provider configurations similar to `testAccAlternateAccountProviderConfig()`, e.g.

```go
func testAccGetAlternateRegion() string {
	v := os.Getenv("AWS_ALTERNATE_REGION")
	if v == "" {
		return "us-east-1"
	}
	return v
}

func testAccAlternateAccountAlternateRegionProviderConfig() string {
	return fmt.Sprintf(`
provider "aws" {
  access_key = %[1]q
  alias      = "alternate"
  profile    = %[2]q
  region     = %[3]q
  secret_key = %[4]q
}
`, os.Getenv("AWS_ALTERNATE_ACCESS_KEY_ID"), os.Getenv("AWS_ALTERNATE_PROFILE"), testAccGetAlternateRegion(), os.Getenv("AWS_ALTERNATE_SECRET_ACCESS_KEY"))
}

func testAccAlternateRegionProviderConfig() string {
	return fmt.Sprintf(`
provider "aws" {
  alias  = "alternate"
  region = %[1]q
}
`, testAccGetAlternateRegion())
}
```

The above should cover most of our acceptance testing use cases while allowing us to use a standard framework for implementations across the provider. 👍

Reply:
@bflad Yes, I only saw those recommendations after making these changes - I'll incorporate the recommendations and enhance where appropriate. Thanks.